packages/turbo-vsc/src/extension.ts

```diff
@@ -0,0 +1,328 @@
+
+  // If the extension is launched in debug mode then the debug server options are used
+  // Otherwise the run options are used
+
+  let lspPath = Uri.joinPath(
+    context.extensionUri,
+    "out",
+    `turborepo-lsp-${process.platform}-${process.arch}`
+  ).fsPath;
+
+  const serverOptions: ServerOptions = {
+    run: {
+      command: lspPath,
+    },
+    debug: {
+      command: lspPath,
+    },
+  };
+
+  // Options to control the language client
+  const clientOptions: LanguageClientOptions = {
+    // Register the server for turbo json documents
+    documentSelector: [
+      { scheme: "file", pattern: "**/turbo.json" },
+      { scheme: "file", pattern: "**/package.json" },
+    ],
+  };
+
+  // Create the language client and start the client.
+  client = new LanguageClient(
+    "turboLSP",
+    "Turborepo Language Server",
+    serverOptions,
+    clientOptions
+  );
+
+  // Start the client. This will also launch the server
+  client.start();
+}
+
+export function deactivate(): Thenable<void> | undefined {
+  if (!client) {
+    return undefined;
+  }
+  return client.stop();
+}
+
+function updateStatusBarItem(running: boolean) {
+  toolbar.command = running ? "turbo.daemon.stop" : "turbo.daemon.start";
+  toolbar.text = running ? `Turbo Running` : "Turbo Stopped";
+  toolbar.show();
+}
+
+function updateJSONDecorations(editor?: TextEditor) {
+  if (
+    !editor ||
+    !path.basename(editor.document.fileName).endsWith("turbo.json")
+  ) {
+    return;
+  }
+
+  let isPipelineKey = false;
+  let pipelineDepth = -1; // indicates we're not in a pipeline block
+
+  const ranges: Range[] = [];
+  visit(editor.document.getText(), {
+    onObjectProperty: (property, offset, length) => {
+      if (property === "pipeline") {
+        isPipelineKey = true;
+        for (let i = 1; i < 9; i++) {
+          let index = i + offset;
+          editor.setDecorations(pipelineColors[i], [
+            new Range(
+              editor.document.positionAt(index),
+              editor.document.positionAt(index + 1)
+            ),
+          ]);
+        }
+      }
+      if (isPipelineKey && pipelineDepth === 0) {
+        ranges.push(
+          new Range(
+            editor.document.positionAt(offset),
+            editor.document.positionAt(offset + length)
+          )
+        );
+      }
+    },
+    onObjectBegin: (offset, length) => {
+      if (isPipelineKey && pipelineDepth === -1) {
+        pipelineDepth = 0;
+      } else if (pipelineDepth !== -1) {
+        pipelineDepth += 1;
+      }
+    },
+    onObjectEnd: (offset, length) => {
+      if (pipelineDepth === 0) {
+        pipelineDepth = -1;
+      } else if (pipelineDepth !== -1) {
+        pipelineDepth -= 1;
+      }
+    },
+  });
+
+  // editor.setDecorations(decoration, ranges);
+}
+
+async function promptGlobalTurbo() {
+  let answer = await window.showErrorMessage(
+    "Turbo CLI not found. Please install Turbo CLI to use this extension.",
+    "Install Now"
```

Comment:
Terminology here is a little inconsistent. Rule of thumb to follow:

Referring to the product should always be `Turborepo`, referring to the CLI should always be `turbo`.

```suggestion
    "turbo not found. See https://turbo.build/repo/docs/installing",
    "Install Now"
```